crates/autopilot/src/boundary/events/cow_amm_product_factory.rs

```diff
@@ -0,0 +1,104 @@
+use {
+    contracts::CowAmmConstantProductFactory,
+    ethcontract::{common::DeploymentInformation, dyns::DynWeb3},
+    ethrpc::current_block::RangeInclusive,
+    primitive_types::H160,
+    shared::{event_handling::EventStoring, impl_event_retrieving},
+    std::{
+        collections::{BTreeMap, HashSet},
+        sync::Arc,
+    },
+    tokio::sync::RwLock,
+};
+
+impl_event_retrieving! {
+    pub CowAmmConstantProductFactoryContract for contracts::cow_amm_constant_product_factory
+}
+
+/// CoW AMM indexer which stores events in-memory.
+#[derive(Clone)]
+pub struct Indexer {
+    /// Registry with the format: (block number, event)
+    registry: Arc<
+        RwLock<BTreeMap<u64, contracts::cow_amm_constant_product_factory::event_data::Deployed>>,
+    >,
+    first_block: u64,
+}
+
+impl Indexer {
+    pub async fn new(web3: &DynWeb3) -> Self {
+        let cow_amm_product_factory = CowAmmConstantProductFactory::deployed(web3)
+            .await
+            .expect("Failed to deploy CowAmmConstantProductFactory");
```

Comment:
We don't deploy anything here we just try to find whatever was deployed.
```suggestion
            .expect("Failed to find deployed CowAmmConstantProductFactory");
```